app/js/history/controllers/HistoryCtrl.js

```diff
@@ -0,0 +1,96 @@
+"use strict";
+/**
+ * @module   FM.history.HistoryCtrl
+ * @author   SOON_
+ * @requires FM.api
+ * @requires ngRoute
+ */
+angular.module("FM.history.HistoryCtrl", [
+    "FM.api.PlayerHistoryResource",
+    "infinite-scroll",
+    "ngRoute"
+])
+/**
+ * @method config
+ * @param  {Provider} $routeProvider
+ */
+.config([
+    "$routeProvider",
+    function ($routeProvider) {
+
+        $routeProvider
+            .when("/history", {
+                templateUrl: "partials/history.html",
+                controller: "HistoryCtrl",
+                resolve: {
+                    historyData: ["PlayerHistoryResource", "$q", "$route", function (PlayerHistoryResource, $q, $route){
+                        var deferred = $q.defer();
+
+                        // resolve headers from query
+                        PlayerHistoryResource.query($route.current.params, function(response, headers){
+                            deferred.resolve({
+                                data: response,
+                                headers: headers
+                            });
```

Comment:
you should avoid putting logic like this in the config, if you need to serialize the data it should be done in the resource using transformResponse function, like with the `playing` flag from the transport resource